lib/shopify_cli/theme/dev_server/hot-reload.js

```diff
@@ -1,93 +1,131 @@
 (() => {
-  function connect() {
-    const eventSource = new EventSource('/hot-reload');
-
-    eventSource.onmessage = handleUpdate;
-
-    eventSource.onopen = () => console.log('[HotReload] SSE connected.');
-
-    eventSource.onclose = () => {
-      console.log('[HotReload] SSE closed. Attempting to reconnect...');
-
-      setTimeout(connect, 5000);
+  class SectionReloadEvent extends Event {
```

Comment:
Generally, relying on `CustomEvent` in this scenario is a good idea (mainly because it brings the `CustomEvent.detail` read-only property, which ensures the event is consistent). But the current version works as well, so feel free to keep it as is :)